src/base/bittorrent/session.cpp

```diff
@@ -3816,22 +3817,30 @@ bool Session::addMoveTorrentStorageJob(TorrentHandleImpl *torrent, const QString
         if (iter != m_moveStorageQueue.end()) {
             // remove existing inactive job
             m_moveStorageQueue.erase(iter);
+            LogMsg(tr("Cancelled to move \"%1\" from \"%2\" to \"%3\".").arg(torrent->name(), currentLocation, iter->path));
         }
     }
 
-    QString currentLocation = torrent->actualStorageLocation();
     if (!m_moveStorageQueue.isEmpty() && (m_moveStorageQueue.first().torrentHandle == torrentHandle)) {
-        // if there is active job for this torrent consider its target path as current location
-        // of this torrent to prevent creating meaningless job that will do nothing
-        currentLocation = m_moveStorageQueue.first().path;
+        // if there is active job for this torrent prevent creating meaningless
+        // job that will move torrent to the same location as current one
+        if (QDir {m_moveStorageQueue.first().path} == QDir {newPath}) {
+            LogMsg(tr("Couldn't enqueue move of \"%1\" to \"%2\". Torrent is currently moving to the same destination location.")
+                   .arg(torrent->name(), newPath));
+            return false;
+        }
+    }
+    else {
+        if (QDir {currentLocation} == QDir {newPath}) {
+            LogMsg(tr("Couldn't enqueue move of \"%1\" from \"%2\" to \"%3\". Both paths point to the same location.")
+                   .arg(torrent->name(), currentLocation, newPath));
+            return false;
+        }
     }
-
-    if (QDir {currentLocation} == QDir {newPath})
-        return false;
 
     const MoveStorageJob moveStorageJob {torrentHandle, newPath, mode};
     m_moveStorageQueue << moveStorageJob;
-    qDebug("Move storage from \"%s\" to \"%s\" is enqueued.", qUtf8Printable(currentLocation), qUtf8Printable(newPath));
+    LogMsg(tr("Enqueued to move \"%1\" from \"%2\" to \"%3\".").arg(torrent->name(), currentLocation, newPath));
```

Comment:
IMO, it's incorrect to have source path in this message since it can be another when job is really started. I'll change it (remove source path) if you don't object.
@Chocobo1, @thalieht?

Reply:
>IMO, it's incorrect to have source path in this message since it can be another when job is really started. 

The log message record the event at the time it triggers, will it cause confusion?

Reply:
>The log message record the event at the time it triggers, will it cause confusion?

From my POV, it will. We enqueue a move regardless of the source path where it will be located when the move is applied so event is "move to some destination path is enqueued".

Reply:
But if you insist I leave it as is.

For me, the main value of all these messages is to track the sequence of events in order to be able to find logical errors in case of anything.